src/Samples/Sarif.WorkItems.Sample/Program.cs

```diff
@@ -5,112 +5,83 @@
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
-using Microsoft.CodeAnalysis.Sarif;
-using Microsoft.CodeAnalysis.Sarif.Baseline;
-using Microsoft.CodeAnalysis.Sarif.WorkItemFiling;
+using System.Threading.Tasks;
 using Newtonsoft.Json;
 
 namespace Sarif.WorkItems.Sample
 {
-    public class Program
+    public partial class Program
     {
         private static int Main(string[] args)
         {
-            NonFunctioningEndToEndApiUsageExample();
-            return 0;
-        }
-
-        private static void NonFunctioningEndToEndApiUsageExample()
-        {
+            // Mock up of end-to-end API use.
+            
             // 1. Retrieve the current SARIF and its baseline equivalent
-            SarifLog baselineSarif = RetrieveBaselineSarif();
-            SarifLog currentSarif = RetrieveCurrentSarif();
+            SarifLog baselineSarif = RetrieveBaselineSarif(args[0]);
+            SarifLog currentSarif = RetrieveCurrentSarif(args[1]);
+            string outputFilePath = args[2];
 
-            // 2. Match the SARIF files
+            // 2. Given a baseline and current sarif, produce a 'matched' log
             ISarifLogBaseliner resultMatcher = SarifLogBaselinerFactory.CreateSarifLogBaseliner(SarifBaselineType.Standard);
 
-            SarifLog matchedLog = new SarifLog
-            {
-                Runs = new[]
-                {
-                    resultMatcher.CreateBaselinedRun(baselineSarif.Runs[0], currentSarif.Runs[0])
-                }
-            };
+            SarifLog matchedSarif = resultMatcher.CreateBaselinedSarifLog(
+                baselineSarif, 
+                currentSarif);
 
-            // 3. Make sure everything has a GUID and that we have a map back to each result from it
-            var guidToMatchedResultMap = new Dictionary<string, Result>();
-            foreach (Result result in matchedLog.Runs[0].Results)
-            {
-                if (result.Guid == null)
-                {
-                    result.Guid = Guid.NewGuid().ToString();
-                }
-                guidToMatchedResultMap[result.Guid] = result;
-            }
+            // 3. We need a map from each result guid to the matched SARIF result.
+            //    As bugs are filed against refactored files, we use the guid to 
+            //    link to the original file in order to populate the WIT URI.
 
-            var adoClient = new AzureDevOpsClient();
-            var filingTarget = new AzureDevOpsFilingTarget(adoClient);
+            var guidMappingVisitor = new GuidMappingVisitor();
+            guidMappingVisitor.Visit(matchedSarif);
 
-            var workItemFiler = new WorkItemFiler(
-                filingTarget,
-                filteringStrategy: new NewResultsFilteringStrategy(),
-                groupingStrategy: new OneResultPerWorkItemGroupingStrategy());
+            // 4. Now we need to refactor the log file into one or more log files, 
```

Comment:
>refactor [](start = 33, length = 8)

factor. Or split.

Reply:
yes.

---
In reply to: [307083636](https://github.com/microsoft/sarif-sdk/pull/1589#discussion_r307083636) [](ancestors = 307083636)